mlrose/fitness.py

```diff
@@ -925,9 +930,9 @@ def evaluate(self, state):
 
         fitness = 0
 
-        for i in range(len(self.edges)):
-            # Check for adjacent nodes of the same color
-            if state[self.edges[i][0]] == state[self.edges[i][1]]:
+        for v, color in enumerate(state):
+            # Check that all adjacent nodes are different color
+            if all(state[neighbor] != color for neighbor in self.neighbors.get(v, set())):
```

Comment:
Another quote from Isbell's paper:
>  Here, we define Max K-Coloring to be the task of finding a coloring that minimizes the number of adjacent pairs colored the same
There's some question in my mind as to whether `all` should be used here or if `fitness` should be proportional to the number of edges for which this holds true. In either case, I noticed that there's a test failing because the fitness value here changed, so this probably shouldn't be merged right now either way.

Reply:
Hi @asantas93,
I think, as you said, the `all` presents a problem here, since what we are interested in isn't whether all the neighbors are a different color from a given node, but how many adjacent pairs are colored the same (as stated in the Isbell quote that you provided). If you could find some way of rewriting line 935 of your code in order to account for that, then I think your code could be potentially a lot faster than the original code for this fitness function.
Regards,
Genevieve.